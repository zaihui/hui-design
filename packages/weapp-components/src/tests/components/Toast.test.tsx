import React from 'react'
import Taro from '@tarojs/taro'
import HuiToast from '@/components/Toast'
import { renderToString } from 'nerv-server'

// 组件里使用了Taro.pxTransform()的话需要在执行测试文件前先调Taro.initPxTransform()
// 因为测试文件是直接引入的组件文件 没有进行编译 所以直接执行测试文件会报错
Taro.initPxTransform({ designWidth: 375, deviceRatio: {} })

describe('huiIcon Snap', () => {
  it('renders HuiToast -- text', () => {
    expect.hasAssertions()
    const component = renderToString(<HuiToast title='Hello' type='text' visible />)
    expect(component).toMatchSnapshot()
  })

  it('renders HuiToast -- success', () => {
    expect.hasAssertions()
    const component = renderToString(<HuiToast title='success' type='success' visible />)
    expect(component).toMatchSnapshot()
  })

  it('renders HuiToast -- fail', () => {
    expect.hasAssertions()
    const component = renderToString(<HuiToast title='fail' type='fail' visible />)
    expect(component).toMatchSnapshot()
  })

  it('renders HuiToast -- custom', () => {
    expect.hasAssertions()
    const component = renderToString(
      <HuiToast title='custom' type='custom' icon='001-close' visible />,
    )
    expect(component).toMatchSnapshot()
  })
})
