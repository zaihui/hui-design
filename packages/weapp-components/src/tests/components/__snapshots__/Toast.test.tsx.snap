// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`huiIcon Snap renders HuiToast -- custom 1`] = `"<div class=\\"hui-toast-box\\"><div class=\\"toast\\"><div class=\\"icon\\"><div style=\\"color:inherit;font-size:1.536rem;\\" class=\\"hui-icons hui-icon-001-close\\"></div></div><span class=\\"taro-text text\\">custom</span></div></div>"`;

exports[`huiIcon Snap renders HuiToast -- fail 1`] = `"<div class=\\"hui-toast-box\\"><div class=\\"toast\\"><div class=\\"icon\\"><div style=\\"color:inherit;font-size:1.536rem;\\" class=\\"hui-icons hui-icon-001-close\\"></div></div><span class=\\"taro-text text\\">fail</span></div></div>"`;

exports[`huiIcon Snap renders HuiToast -- success 1`] = `"<div class=\\"hui-toast-box\\"><div class=\\"toast\\"><div class=\\"icon\\"><div style=\\"color:inherit;font-size:1.536rem;\\" class=\\"hui-icons hui-icon-003-right\\"></div></div><span class=\\"taro-text text\\">success</span></div></div>"`;

exports[`huiIcon Snap renders HuiToast -- text 1`] = `"<div class=\\"hui-toast-box\\"><div class=\\"toast\\">undefined<span class=\\"taro-text text\\">Hello</span></div></div>"`;
